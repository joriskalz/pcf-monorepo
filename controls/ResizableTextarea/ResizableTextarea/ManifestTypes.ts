export enum BorderStyles {
  Normal = 0,
  Centred = 1,
}

export const ResizeTypes: Record<"0" | "1" | "2" | "3", string> = {
  "0": "none",
  "1": "both",
  "2": "vertical",
  "3": "horizontal",
};

export interface InputParameters {
  DefaultHeight?: number | null;
  DefaultWidth?: number | null;
  InputEvent?: string | null;
  Text: string;
  Default: string;
}

export interface OutputParameterFlags {
  resizedHeight?: boolean;
  resizedWidth?: boolean;
  text?: boolean;
}

export const enum InputEvents {
  SetFocus = "SetFocus",
  ResetSize = "ResetSize",
  ResetWidth = "ResetWidth",
  ResetHeight = "ResetHeight",
  ResetText = "ResetText",
}

export const enum FontSizeUnits {
  Pt = "0",
  Px = "1",
}
