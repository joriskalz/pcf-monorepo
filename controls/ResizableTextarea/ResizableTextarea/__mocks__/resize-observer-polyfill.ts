/* istanbul ignore file */

/* eslint-disable @typescript-eslint/no-explicit-any */
/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
/* eslint-disable @typescript-eslint/no-unused-vars */

export const ResizeObserverMock = {
  onResize: jest.fn(),
  observe: jest.fn(),
  unobserve: jest.fn(),
  disconnect: jest.fn(),
};
export default class ResizeObserver {
  constructor(handler: any) {
    ResizeObserverMock.onResize = handler;
    this.observe = ResizeObserverMock.observe;
    this.unobserve = ResizeObserverMock.unobserve;
    this.disconnect = ResizeObserverMock.disconnect;
  }
  disconnect(): void;
  disconnect(): void;
  disconnect(): void {
    throw new Error();
  }
  observe(target: Element, options?: ResizeObserverOptions): void;
  observe(target: Element): void;
  observe(target: any, options?: any): void {
    throw new Error();
  }
  unobserve(target: Element): void;
  unobserve(target: Element): void;
  unobserve(target: any): void {
    throw new Error();
  }
}
