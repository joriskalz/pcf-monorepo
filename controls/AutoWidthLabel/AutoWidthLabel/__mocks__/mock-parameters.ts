/* istanbul ignore file */

import { IInputs } from "../generated/ManifestTypes";
import {
  MockDecimalNumberProperty,
  MockEnumProperty,
  MockStringProperty,
  MockWholeNumberProperty,
} from "./mock-context";

export function getMockParameters(): IInputs {
  return {
    Text: new MockStringProperty(),
    PaddingLeft: new MockWholeNumberProperty(),
    PaddingRight: new MockWholeNumberProperty(),
    PaddingTop: new MockWholeNumberProperty(),
    PaddingBottom: new MockWholeNumberProperty(),
    FontName: new MockStringProperty(),
    FontSize: new MockDecimalNumberProperty(),
    FontSizeUnits: new MockEnumProperty("1"),
    FontColor: new MockStringProperty(),
    FontWeight: new MockStringProperty(),
    DisabledFontColor: new MockStringProperty(),
    DisabledFontWeight: new MockStringProperty(),
    FocusFontColor: new MockStringProperty(),
    FocusFontWeight: new MockStringProperty(),
    HoverFontColor: new MockStringProperty(),
    HoverFontWeight: new MockStringProperty(),
    FillColor: new MockStringProperty(),
    DisabledFillColor: new MockStringProperty(),
    FocusFillColor: new MockStringProperty(),
    HoverFillColor: new MockStringProperty(),
    //DisabledHoverFillColor: new MockStringProperty(),
    BorderColor: new MockStringProperty(),
    BorderThickness: new MockWholeNumberProperty(),
    BorderRadius: new MockWholeNumberProperty(),
    DisabledBorderColor: new MockStringProperty(),
    FocusBorderColor: new MockStringProperty(),
    FocusBorderThickness: new MockWholeNumberProperty(),
    HoverBorderColor: new MockStringProperty(),
    HoverBorderThickness: new MockWholeNumberProperty(),
    //RenderBorderStyle: new MockEnumProperty<"0" | "1">(),
  };
}
