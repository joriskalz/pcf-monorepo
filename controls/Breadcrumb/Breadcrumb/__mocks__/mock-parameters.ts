/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockStringProperty, MockWholeNumberProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        Theme: new MockStringProperty(),
        items: new MockDataSet([]),
        MaxDisplayedItems: new MockWholeNumberProperty(),
        OverflowIndex: new MockWholeNumberProperty(),
    };
}
