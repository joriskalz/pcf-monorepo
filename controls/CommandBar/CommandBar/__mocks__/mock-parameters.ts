/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockStringProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        Theme: new MockStringProperty(),
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        items: new MockDataSet([]),
    };
}
