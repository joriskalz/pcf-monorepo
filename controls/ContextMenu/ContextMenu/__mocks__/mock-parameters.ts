/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockEnumProperty, MockStringProperty, MockTwoOptionsProperty, MockWholeNumberProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        Theme: new MockStringProperty(),
        TextAlignment: new MockEnumProperty(),
        items: new MockDataSet([]),
        Chevron: new MockTwoOptionsProperty(),
        FillColor: new MockStringProperty(),
        BorderColor: new MockStringProperty(),
        BorderRadius: new MockWholeNumberProperty(),
        FontColor: new MockStringProperty(),
        FontSize: new MockWholeNumberProperty(),
        HoverBorderColor: new MockStringProperty(),
        HoverFillColor: new MockStringProperty(),
        HoverFontColor: new MockStringProperty(),
        HoverIconColor: new MockStringProperty(),
        IconColor: new MockStringProperty(),
        IconSize: new MockWholeNumberProperty(),
    };
}
