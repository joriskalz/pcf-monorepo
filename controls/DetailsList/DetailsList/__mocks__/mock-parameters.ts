/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockEnumProperty, MockStringProperty, MockTwoOptionsProperty, MockWholeNumberProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        Theme: new MockStringProperty(),
        Compact: new MockTwoOptionsProperty(),
        CurrentSortColumn: new MockStringProperty(),
        CurrentSortDirection: new MockEnumProperty('0'),
        PageSize: new MockWholeNumberProperty(),
        RaiseOnRowSelectionChangeEvent: new MockTwoOptionsProperty(),
        SelectRowsOnFocus: new MockTwoOptionsProperty(),
        SelectionType: new MockEnumProperty('0'),
        AlternateRowColor: new MockStringProperty(),
        HeaderVisible: new MockTwoOptionsProperty(false),
        SelectionAlwaysVisible: new MockTwoOptionsProperty(false),
        records: new MockDataSet([]),
        columns: new MockDataSet([]),
        LargeDatasetPaging: new MockTwoOptionsProperty(),
    };
}
