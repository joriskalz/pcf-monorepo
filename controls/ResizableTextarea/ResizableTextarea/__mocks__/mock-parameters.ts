/* istanbul ignore file */

import { IInputs } from "../generated/ManifestTypes";
import {
  MockDecimalNumberProperty,
  MockEnumProperty,
  MockStringProperty,
  MockWholeNumberProperty,
} from "./mock-context";

export function getMockParameters(): IInputs {
  return {
    Text: new MockStringProperty(),
    Default: new MockStringProperty(),
    MaxLength: new MockWholeNumberProperty(),
    HintText: new MockStringProperty(),
    EmptyPlaceholderText: new MockStringProperty(),
    ResizedHeight: new MockWholeNumberProperty(),
    ResizedWidth: new MockWholeNumberProperty(),
    DefaultHeight: new MockWholeNumberProperty(),
    DefaultWidth: new MockWholeNumberProperty(),
    MinHeight: new MockWholeNumberProperty(),
    MaxHeight: new MockWholeNumberProperty(),
    MinWidth: new MockWholeNumberProperty(),
    MaxWidth: new MockWholeNumberProperty(),
    PaddingLeft: new MockWholeNumberProperty(),
    PaddingRight: new MockWholeNumberProperty(),
    PaddingTop: new MockWholeNumberProperty(),
    PaddingBottom: new MockWholeNumberProperty(),
    LeftPaddingAdjustment: new MockWholeNumberProperty(),
    AllowResize: new MockEnumProperty<"0" | "1" | "2" | "3">(),
    EnableSpellCheck: new MockEnumProperty<"0" | "1">(),
    FontName: new MockStringProperty(),
    FontSize: new MockDecimalNumberProperty(),
    FontSizeUnits: new MockEnumProperty(),
    FontColor: new MockStringProperty("1"),
    FontWeight: new MockStringProperty(),
    DisabledFontColor: new MockStringProperty(),
    DisabledFontWeight: new MockStringProperty(),
    FocusFontColor: new MockStringProperty(),
    FocusFontWeight: new MockStringProperty(),
    HoverFontColor: new MockStringProperty(),
    HoverFontWeight: new MockStringProperty(),
    HintFontColor: new MockStringProperty(),
    HintFontWeight: new MockStringProperty(),
    FillColor: new MockStringProperty(),
    DisabledFillColor: new MockStringProperty(),
    FocusFillColor: new MockStringProperty(),
    HoverFillColor: new MockStringProperty(),
    DisabledHoverFillColor: new MockStringProperty(),
    BorderColor: new MockStringProperty(),
    BorderThickness: new MockWholeNumberProperty(),
    BorderRadius: new MockWholeNumberProperty(),
    DisabledBorderColor: new MockStringProperty(),
    FocusBorderColor: new MockStringProperty(),
    FocusBorderThickness: new MockWholeNumberProperty(),
    HoverBorderColor: new MockStringProperty(),
    HoverBorderThickness: new MockWholeNumberProperty(),
    RenderBorderStyle: new MockEnumProperty<"0" | "1">(),
    AccessibilityLabel: new MockStringProperty(),
    InputEvent: new MockStringProperty(),
  };
}
