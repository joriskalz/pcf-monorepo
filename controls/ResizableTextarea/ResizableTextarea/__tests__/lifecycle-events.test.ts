import { ResizableTextarea } from "..";
import { IInputs } from "../generated/ManifestTypes";
import { MockContext, MockState } from "../__mocks__/mock-context";
import { getMockParameters } from "../__mocks__/mock-parameters";
import { ResizeObserverMock } from "../__mocks__/resize-observer-polyfill";

const LIPSUM = "Lorem Ipsum";
const LIPSUM_EDIT = "Lorem Ipsum Edit";

describe("ResizeableTextArea lifecycle events", () => {
  beforeEach(() => jest.clearAllMocks());
  it("initialises ResizeObserver on init, and disconnects on destroy", () => {
    const { component, context, notifyOutputChanged, state, container } = createComponent();
    component.init(context, notifyOutputChanged, state, container);
    expect(ResizeObserverMock.observe).toHaveBeenCalledTimes(1);

    component.destroy();
    expect(ResizeObserverMock.disconnect).toHaveBeenCalledTimes(1);
  });

  it("does not call notifyOutputChanged inside init", () => {
    const { component, context, notifyOutputChanged, state, container } = createComponent();
    component.init(context, notifyOutputChanged, state, container);
    // Check that the notify event has not been called
    expect(notifyOutputChanged).toHaveBeenCalledTimes(0);
  });

  it("creates the textarea element wrapped in containers with text value", () => {
    const textValue = LIPSUM;
    const { component, context, notifyOutputChanged, state, container } = createComponent();
    context.parameters.Text.raw = textValue;
    component.init(context, notifyOutputChanged, state, container);

    expect(container).toMatchSnapshot("Container after init");
    expect(container.getElementsByTagName("textarea")[0].value).toBe(textValue);
  });

  it("creates the textarea element wrapped in containers with default value", () => {
    const textValue = LIPSUM;
    const { component, context, notifyOutputChanged, state, container } = createComponent();
    context.parameters.Default.raw = textValue;
    context.parameters.Text.raw = "foo";
    component.init(context, notifyOutputChanged, state, container);
    expect(container.getElementsByTagName("textarea")[0].value).toBe(textValue);
  });

  it("updates textarea value when host provides updated text", () => {
    const textValue = LIPSUM;
    const newTextValue = LIPSUM_EDIT;
    const { component, context, notifyOutputChanged, state, container } = createComponent();
    context.parameters.Text.raw = textValue;
    component.init(context, notifyOutputChanged, state, container);

    // Set the new text value and set the updateProperties
    context.parameters.Text.raw = newTextValue;
    context.updatedProperties = ["Text", "Parameters"];
    component.updateView(context);
    expect(container.getElementsByTagName("textarea")[0].value).toBe(newTextValue);
  });

  it("updates textarea value when host provides new default", () => {
    const textValue = LIPSUM;
    const newTextValue = LIPSUM_EDIT;
    const newDefaultValue = LIPSUM_EDIT;
    const { component, context, notifyOutputChanged, state, container } = createComponent();
    context.parameters.Text.raw = textValue;
    component.init(context, notifyOutputChanged, state, container);

    // Set the new text value and set the updateProperties
    context.parameters.Text.raw = newTextValue;
    context.updatedProperties = ["Text", "Parameters"];
    component.updateView(context);

    // Set the new Default value and set the updateProperties
    context.parameters.Default.raw = newDefaultValue;
    context.updatedProperties = ["Default", "Parameters"];
    component.updateView(context);
    expect(container.getElementsByTagName("textarea")[0].value).toBe(newDefaultValue);
  });

  it("updates font styles", () => {
    const { component, context, notifyOutputChanged, state, container } = createComponent();
    component.init(context, notifyOutputChanged, state, container);

    expect(container).toMatchSnapshot();

    // Set the new Default value and set the updateProperties
    context.parameters.FontSize.raw = 10;
    component.updateView(context);
    expect(container).toMatchSnapshot("Container after FontSize update");
  });

  it("updates FontSize", () => {
    const textValue = "Lorem Ipsum";

    const { component, context, notifyOutputChanged, state, container } = createComponent();
    context.parameters.Text.raw = textValue;
    component.init(context, notifyOutputChanged, state, container);
    context.parameters.FontSize.raw = 10;
    expect(container).toMatchSnapshot("initial");

    // Set the new Default value and set the updateProperties
    context.parameters.FontSize.raw = 20;
    component.updateView(context);
    expect(container).toMatchSnapshot("after fontsize update");
  });

  it("sets positive tabindex for canvas apps", () => {
    const { component, context, notifyOutputChanged, state, container } = createComponent();
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    (context as any).accessibility = { assignedTabIndex: 15 };
    component.init(context, notifyOutputChanged, state, container);
    expect(container).toMatchSnapshot();
  });

  it("resets text to default when ResetText event is provided", () => {
    const newTextValue = LIPSUM_EDIT;
    const defaultValue = LIPSUM;
    const { component, context, notifyOutputChanged, state, container } = createComponent();
    context.parameters.Default.raw = defaultValue;
    component.init(context, notifyOutputChanged, state, container);
    expect(container.getElementsByTagName("textarea")[0].value).toBe(defaultValue);

    // Set the new text value and set the updateProperties
    context.parameters.Text.raw = newTextValue;
    context.updatedProperties = ["Text", "Parameters"];
    component.updateView(context);
    expect(container.getElementsByTagName("textarea")[0].value).toBe(newTextValue);

    // Set the new Default value and set the updateProperties
    context.parameters.InputEvent.raw = "ResetText_123";
    context.updatedProperties = ["Default", "Parameters"];
    component.updateView(context);
    expect(container.getElementsByTagName("textarea")[0].value).toBe(defaultValue);
  });
});

function createComponent() {
  const component = new ResizableTextarea();
  const notifyOutputChanged = jest.fn();
  const context = new MockContext<IInputs>(getMockParameters());
  const state = new MockState();
  const container = document.createElement("div");
  document.body.appendChild(container);
  return { component, context, notifyOutputChanged, state, container };
}
