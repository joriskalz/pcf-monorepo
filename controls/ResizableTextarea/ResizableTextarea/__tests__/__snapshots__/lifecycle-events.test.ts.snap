// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResizeableTextArea lifecycle events creates the textarea element wrapped in containers with text value: Container after init 1`] = `
<div>
  <div
    class="cat-textarea-outer"
  >
    <div
      class="cat-textarea-border-container"
      style="border-width: 0px;"
    >
      <div
        class="cat-textarea-border-inner"
        style="margin: 0px;"
      >
        <textarea
          maxlength="100000"
          style="resize: both;"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`ResizeableTextArea lifecycle events sets positive tabindex for canvas apps 1`] = `
<div>
  <div
    class="cat-textarea-outer"
  >
    <div
      class="cat-textarea-border-container"
      style="border-width: 0px;"
    >
      <div
        class="cat-textarea-border-inner"
        style="margin: 0px;"
      >
        <textarea
          maxlength="100000"
          style="resize: both;"
          tabindex="15"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`ResizeableTextArea lifecycle events updates FontSize: after fontsize update 1`] = `
<div>
  <div
    class="cat-textarea-outer"
  >
    <div
      class="cat-textarea-border-container"
      style="border-width: 0px;"
    >
      <div
        class="cat-textarea-border-inner"
        style="margin: 0px;"
      >
        <textarea
          maxlength="100000"
          style="resize: both; font-size: 20px;"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`ResizeableTextArea lifecycle events updates FontSize: initial 1`] = `
<div>
  <div
    class="cat-textarea-outer"
  >
    <div
      class="cat-textarea-border-container"
      style="border-width: 0px;"
    >
      <div
        class="cat-textarea-border-inner"
        style="margin: 0px;"
      >
        <textarea
          maxlength="100000"
          style="resize: both;"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`ResizeableTextArea lifecycle events updates font styles 1`] = `
<div>
  <div
    class="cat-textarea-outer"
  >
    <div
      class="cat-textarea-border-container"
      style="border-width: 0px;"
    >
      <div
        class="cat-textarea-border-inner"
        style="margin: 0px;"
      >
        <textarea
          maxlength="100000"
          style="resize: both;"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`ResizeableTextArea lifecycle events updates font styles: Container after FontSize update 1`] = `
<div>
  <div
    class="cat-textarea-outer"
  >
    <div
      class="cat-textarea-border-container"
      style="border-width: 0px;"
    >
      <div
        class="cat-textarea-border-inner"
        style="margin: 0px;"
      >
        <textarea
          maxlength="100000"
          style="resize: both; font-size: 10px;"
        />
      </div>
    </div>
  </div>
</div>
`;
